### ISSUE TITLE
builds when changes made to PR and a list of branches

### ISSUE THREAD

**Issue by mengqing** [jenkinsci/ghprb-plugin#154]
What configuration would I need in order for changes to PR and a list of branches to trigger builds?

Currently I've got +refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/* in the Refspec and ${sha1} in the Branches to build.

It builds when changes to PR are made, however builds not triggered when changes to eg master branch are made

**Comment by KostyaSha** [comment]
It's git plugin and jenkins design, or probably ghprb. You can't use PR job for regular builds.

**Comment by mengqing** [comment]
So I can't have both ghprb and github-plugin running together?

**Comment by KostyaSha** [comment]
Yes

**Comment by DavidTanner** [comment]
You would need to add another trigger to the job.  The ghprb trigger is only going to run on plugins, then the source poll trigger could run on changes to your branches.

**Comment by mengqing** [comment]
@DavidTanner At the moment, I've got GitHub Pull Request Builder and Build when a change is pushed to GitHub triggers enabled, would this be sufficient?

If not, what other triggers can I use to do what I want to do?

**Comment by arcivanov** [comment]
Simply generate multiple jobs with different trigger versions automatically using jenkinsci/job-dsl. Create a job template and only change triggers, branch designators and publishers. You also want to make sure not to publish artifacts in PR jobs vs artifacts from stable branches.

**Comment by benpatterson** [comment]
As @arcivanov mentioned, this plugin is designed for pull requests, and what you're asking for--while totally reasonable--is simply out of scope. I'm going to close this for now.

**Comment by guruvamsichintala** [comment]
With git webhook configured , How do I build a job for specific branch when someone made a push to it. I have enabled trigger when pushed code to github.  Right now all jenkins jobs are triggered even though it was pushed for a different branch.

**Comment by benpatterson** [comment]
@guruvamsichintala you should be able to whitelist branches on the job configuration. However, if you have multiple jobs, you'd need to set that configuration in each job. HTH

**Comment by guruvamsichintala** [comment]
I have already specified branch name in "Branches to be build" option. But still when I push code to any branch , all the jobs which are configured for specific branch and triggered when something pushed to git are running.  I figured it out with 

> http://stackoverflow.com/questions/5784329/how-can-i-make-jenkins-ci-with-git-trigger-on-pushes-to-master/9921495#9921495

Configured git plugin inside git, specified the url inside plugin and configured job with poll scm (no schedule).  Worked perfectly
